ios/FluentUI/Bottom Sheet/BottomSheetController.swift

```diff
@@ -0,0 +1,371 @@
+}
+
+@objc(MSFBottomSheetController)
+public class BottomSheetController: UIViewController {
+
+    /// Initializes the bottom sheet controller.
+    /// - Parameter contentViewController: The view controller that's placed inside the bottom sheet.
+    ///
+    /// By default the root view of `contentViewController` will be sized automatically to fill the available area,
+    /// respecting the provided `preferredExpandedHeightFraction` multiplier.
+    /// Alternatively, the content can size itself by setting `respectsPreferredContentSize` to true and providing a `preferredContentSize`.
+    @objc public init(with contentViewController: UIViewController) {
+        self.contentViewController = contentViewController
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    @available(*, unavailable)
+    required init?(coder: NSCoder) {
+        preconditionFailure("init(coder:) has not been implemented")
+    }
+
+    /// A scroll view in `contentViewController`'s view hierarchy.
+    /// Provide this to ensure the bottom sheet pan gesture recognizer coordinates with the scroll view to enable scrolling based on current bottom sheet position and content offset.
+    @objc open var hostedScrollView: UIScrollView?
+
+    /// Indicates if the bottom sheet is expandable.
+    @objc open var isExpandable: Bool = true {
+        didSet {
+            if isExpandable != oldValue {
+                resizingHandleView.isHidden = !isExpandable
+                panGestureRecognizer.isEnabled = isExpandable
+                move(to: .collapsed, animated: false)
+            }
+        }
+    }
+
+    /// Indicates if `preferredContentSize` of `contentViewController` should be respected.
+    /// Regardless of the value, the expanded height is limited by the height of `BottomSheetController`'s root view.
+    @objc open var respectsPreferredContentSize: Bool = false {
+        didSet {
+            if respectsPreferredContentSize != oldValue {
+                updateBottomSheetHeightConstraints()
+            }
+        }
+    }
+
+    /// Fraction of the available area that the bottom sheet shold take up in the expanded position.
+    ///
+    /// Ignored when `respectsPreferredContentSize` is set to `true`
+    @objc open var expandedHeightFraction: CGFloat = 1.0 {
+        didSet {
+            if expandedHeightFraction != oldValue && !respectsPreferredContentSize {
+                updateBottomSheetHeightConstraints()
+            }
+        }
+    }
+
+    /// Height of the top portion of the content view that should be visible when the bottom sheet is collapsed.
+    @objc open var collapsedContentHeight: CGFloat = Constants.defaultCollapsedContentHeight
+
+    /// The object that acts as the delegate of the bottom sheet.
+    @objc open weak var delegate: BottomSheetControllerDelegate?
+
+    // MARK: - View loading
+
+    public override func loadView() {
```

Comment:
nit for future reference can you write high-level UIView hierarchy of passthrough view, bottomsheetview, scrollview